crates/biome_js_analyze/src/lint/nursery/no_secrets.rs

```diff
@@ -155,119 +223,284 @@ struct SensitivePattern {
     },
     SensitivePattern {
         pattern: Pattern::Regex(&GOOGLE_SERVICE_ACCOUNT_REGEX),
         comment: "Google (GCP) Service-account",
         min_len: 14,
+        allows_spaces: true,
     },
     SensitivePattern {
         pattern: Pattern::Regex(&TWILIO_API_KEY_REGEX),
         comment: "Twilio API Key",
         min_len: 32,
+        allows_spaces: false,
     },
     SensitivePattern {
         pattern: Pattern::Contains("-----BEGIN RSA PRIVATE KEY-----"),
         comment: "RSA Private Key",
         min_len: 64,
+        allows_spaces: true,
     },
     SensitivePattern {
         pattern: Pattern::Contains("-----BEGIN OPENSSH PRIVATE KEY-----"),
         comment: "SSH (OPENSSH) Private Key",
         min_len: 64,
+        allows_spaces: true,
     },
     SensitivePattern {
         pattern: Pattern::Contains("-----BEGIN DSA PRIVATE KEY-----"),
         comment: "SSH (DSA) Private Key",
         min_len: 64,
+        allows_spaces: true,
     },
     SensitivePattern {
         pattern: Pattern::Contains("-----BEGIN EC PRIVATE KEY-----"),
         comment: "SSH (EC) Private Key",
         min_len: 64,
+        allows_spaces: true,
     },
     SensitivePattern {
         pattern: Pattern::Contains("-----BEGIN PGP PRIVATE KEY BLOCK-----"),
         comment: "PGP Private Key Block",
         min_len: 64,
+        allows_spaces: true,
     },
 ];
+const MIN_PATTERN_LEN: usize = 14;
+
+// Known safe patterns start here
+static BASE64_REGEX: LazyLock<Regex> = LazyLock::new(|| {
+    Regex::new(r"^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$").unwrap()
+});
+static URL_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r"^https?://[a-zA-Z0-9.-]+(/[a-zA-Z0-9./_-]*)?$").unwrap());
+static RELATIVE_PATH_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r"^(?:\.\./|\./|[a-zA-Z0-9_-]+)/?$").unwrap());
+static UNIX_PATH_REGEX: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"^(/[^/]+)+/?$").unwrap());
+static WINDOWS_PATH_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r"^[a-zA-Z]:\\(?:[^\\]+\\?)*$").unwrap());
+static EMAIL_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$").unwrap());
+static PHONE_REGEX: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"^\+?[1-9]\d{1,14}$").unwrap()); // E.164 format
+
+// Combine all known safe patterns into a single list
+static KNOWN_SAFE_PATTERNS: &[&LazyLock<Regex>] = &[
+    &BASE64_REGEX,
+    &URL_REGEX,
+    &RELATIVE_PATH_REGEX,
+    &UNIX_PATH_REGEX,
+    &WINDOWS_PATH_REGEX,
+    &EMAIL_REGEX,
+    &PHONE_REGEX,
+];
 
-const MIN_PATTERN_LEN: usize = 12;
+fn is_known_safe_pattern(data: &str) -> bool {
+    for pattern in KNOWN_SAFE_PATTERNS {
+        if pattern.is_match(data) {
+            return true;
+        }
+    }
+    false
+}
+
+fn detect_secret(data: &str, entropy_threshold: &u16) -> Option<&'static str> {
+    let tokens: Vec<&str> = split_into_tokens(data);
```

Comment:
We could directly call split/filter here and avoid allocating a `Vec` by iterating over the iterator.
Another possibility is to keep `split_into_tokens` and returning `impl Iterator<Item = str>` instead of `Vec<&str>`.
```suggestion
    let tokens = data.split(...).filter(...);
```

Reply:
Keeping the former option because the other refactors have reduced the code to 3 lines 😂 👌